apps/web/middleware.ts

```diff
@@ -0,0 +1,6 @@
+import { withAuth } from "next-auth/middleware";
+import { pages } from "./server/auth";
+
+export default withAuth({
+  pages,
```

Comment:
I had to extract `pages` into separate variable and pass it again in middleware to avoid infinite redirect as mentioned in NextAuth documentation.

https://next-auth.js.org/configuration/nextjs#pages

<img width="711" alt="image" src="https://github.com/user-attachments/assets/a157371f-d433-44f6-856d-9c40257a86b2" />